types/events.go

```diff
@@ -178,7 +191,7 @@ func (a Attribute) String() string {
 
 // ToKVPair converts an Attribute object into a Tendermint key/value pair.
 func (a Attribute) ToKVPair() abci.EventAttribute {
-	return abci.EventAttribute{Key: toBytes(a.Key), Value: toBytes(a.Value)}
+	return abci.EventAttribute{Key: []byte(a.Key), Value: []byte(a.Value)}
```

Comment:
This measurably improved performance!

However its still silly that we have to keep re-allocating heap space for the keys, when ~all the time, they are actually constants. 

I think we should change the data-backend here to be using `[]byte`, at least until abci events use strings.